### PR TITLE
chore: create discussion forms

### LINKED ISSUE #19810: Create discussion category forms
### Describe the proposed change(s).

We already have issue forms. Now we can have discussion forms! [^blog] [^docs]🥳 

Before I go off and create discussion forms for us, let's discuss the requirements. 😉 

#### Discussion categories

Discussions can fall into these categories:

- Ideas & General Discussion
- Mend Renovate App Support
- Polls (forms don't work for these)
- Questions & Answers

#### Copy/paste most of bug report for Questions & Answers?

We can probably copy/paste most of our bug report form for the _Questions & Answers_ discussion form? We probably always want to know things like Renovate version and platform, debug logs, and a general description, and if things broke recently (regression).

#### Keep things free-form for other categories?

The other discussion categories may need to be more free-form, as one form won't fit all use-cases.

#### Other things to put in the forms?

Do you find you always need to ask for `x`, or remind users to do `y`? Those things can be captured in the forms. 😉 

[^blog]: https://github.blog/2023-01-09-github-discussions-just-got-better-with-category-forms/
[^docs]: https://docs.github.com/en/discussions/managing-discussions-for-your-community/creating-discussion-category-forms

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Create discussion forms for:
  - Ideas & General Discussion
  - Mend Renovate App Support 
  - Questions & Answers

## Context

The Poll discussion type has no form, because that's not supported by GitHub. 😄 

Closes #19810.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: .github/DISCUSSION_TEMPLATE/mend-renovate-app-support.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    id: minimal-reproduction-repository
+    attributes:
+      label: Have you created a minimal reproduction repository?
+      description: Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.
+      options:
+        - 'No reproduction repository'
+        - 'No reproduction, but I have linked to a public repo where it occurs'
+        - 'I have linked to a minimal reproduction repository in the bug description'
+    validations:
+      required: true
```
Comment (rarkins) [inline .github/DISCUSSION_TEMPLATE/mend-renovate-app-support.yml:61]: I think let's remove this part, as the quantity of people ticking yes is likely to be very low

File: .github/DISCUSSION_TEMPLATE/questions-answers.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    id: minimal-reproduction-repository
+    attributes:
+      label: Have you created a minimal reproduction repository?
+      description: Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.
+      options:
+        - 'No reproduction repository'
+        - 'No reproduction, but I have linked to a public repo where it occurs'
+        - 'I have linked to a minimal reproduction repository in the bug description'
+    validations:
+      required: true
```
Comment (rarkins) [inline .github/DISCUSSION_TEMPLATE/questions-answers.yml:113]: Also remove

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — DISMISSED** [review]

**Comment by rarkins** [comment]
One question: if we adopt these templates, but need to convert issues to discussions, do they bypass these templates?

**Comment by HonkingGoose** [comment]
There are some problems with my YAML. I'll go fix those first.

File: .github/DISCUSSION_TEMPLATE/ideas-general-discussion.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+body:
```
Comment (HonkingGoose) [inline .github/DISCUSSION_TEMPLATE/ideas-general-discussion.yml:1]: The filename of the discussion forms must match the category names in the discussion tab on the repository.

This means that if you change the name of the discussion, you should also rename the corresponding template file.

**Review by HonkingGoose — COMMENTED** [review]
I think we're good to merge, just wanted to highlight a gotcha. 😉

**Review by rarkins — APPROVED** [review]

**Pull request merged**